engine/src/main/java/org/terasology/rendering/logic/MeshRenderer.java

```diff
@@ -203,8 +202,7 @@ private void renderEntitiesByMaterial(SetMultimap<Material, EntityRef> meshByMat
                     location.getWorldPosition(worldPos);
                     float worldScale = location.getWorldScale();
 
-                    javax.vecmath.Matrix4f matrixWorldSpace = new javax.vecmath.Matrix4f(VecMath.to(worldRot), VecMath.to(worldPos), worldScale);
-                    transWorldSpace.set(matrixWorldSpace);
+                    Transform transWorldSpace = new Transform(worldPos, worldRot, worldScale);
 
                     Vector3f worldPositionCameraSpace = new Vector3f();
```

Comment:
I am aware you are inheriting this, but could this be renamed as `offsetFromCamera`?

Reply:
I can. Nice catch.